src/main/java/org/iac2/service/architecturereconstruction/plugin/implementation/mysql/MySqlDbModelRefinementPlugin.java

```diff
@@ -0,0 +1,137 @@
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+
+import io.github.edmm.model.DeploymentModel;
+import io.github.edmm.model.component.RootComponent;
+import io.github.edmm.model.relation.HostedOn;
+import org.iac2.common.PluginDescriptor;
+import org.iac2.common.exception.PropertyMissingException;
+import org.iac2.common.model.InstanceModel;
+import org.iac2.common.model.ProductionSystem;
+import org.iac2.common.utility.Edmm;
+import org.iac2.service.architecturereconstruction.common.interfaces.ModelRefinementPlugin;
+import org.iac2.service.architecturereconstruction.common.model.EdmmTypes.MySqlDb;
+import org.iac2.service.architecturereconstruction.common.model.EdmmTypes.MySqlDbms;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// see: https://serverfault.com/questions/263868/how-to-know-all-the-users-that-can-access-a-database-mysql
+// see: https://mkyong.com/regular-expressions/how-to-validate-ip-address-with-regular-expression/
+// see: https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-usagenotes-connect-drivermanager.html
+public class MySqlDbModelRefinementPlugin implements ModelRefinementPlugin {
+    public static final String CONFIG_ENTRY_IGNORE_MISSING_PROPERTIES = "ignoreMissingProperties";
+    public static final String EDMM_PROPERTY_NAME_USERS = "users";
+    private static final Logger LOGGER = LoggerFactory.getLogger(MySqlDbModelRefinementPlugin.class);
+    private final MySqlDbModelRefinementPluginDescriptor descriptor;
+    private boolean ignoreMissingProperties;
+
+    public MySqlDbModelRefinementPlugin(MySqlDbModelRefinementPluginDescriptor descriptor) {
+        this.descriptor = descriptor;
+    }
+
+    public static Collection<String> getUsersOfDatabase(String ip, String port, String user, String password, String database) throws SQLException {
+        final String connectionString = String.format("jdbc:mysql://%s:%s/mysql?user=%s&password=%s", ip, port, user, password);
+        Collection<String> result = new ArrayList<>();
+
+        try (Connection conn = DriverManager.getConnection(connectionString)) {
+            final String statementSQL = """
+                     SELECT user FROM mysql.db WHERE db='%s'
+                     UNION
+                     SELECT user FROM mysql.tables_priv WHERE db='%s'
+                     UNION
+                     SELECT user FROM mysql.columns_priv WHERE db='%s'
+                     UNION
+                     SELECT user FROM mysql.procs_priv WHERE db='%s'
+                    """.formatted(database, database, database, database);
+            try (Statement statement = conn.createStatement()) {
+                ResultSet resultSet = statement.executeQuery(statementSQL);
+
+                while (resultSet.next()) {
+                    result.add(resultSet.getString(1));
```

Comment:
.getString(1) just smells like it will break, but it probably has to be like that, right? :D

Reply:
so the statements select only one column (user) and for some reason mysql (or is it SQL?) uses 1-based indices.
So `getString(1)` gets the value of the first (and only) column, which shouldn't break easily